packages/react-components/src/components/CaptionsSettingModal.tsx

```diff
@@ -80,27 +81,37 @@ export const _CaptionsSettingModal = (props: _CaptionsSettingModalProps): JSX.El
   const localeStrings = useLocale().strings.captionsSettingModal;
   const strings = props.strings ?? localeStrings;
 
-  const onDismissTriggered = (): void => {
+  const onDismissTriggered = useCallback((): void => {
     if (onDismissCaptionsSetting) {
       onDismissCaptionsSetting();
     }
-  };
+  }, [onDismissCaptionsSetting]);
 
-  const onClickConfirm = (language: string): void => {
-    if (isCaptionsFeatureActive) {
-      onSetSpokenLanguage(language);
-    } else {
-      onStartCaptions({ spokenLanguage: selectedItem.text });
-    }
-    onDismissTriggered();
-  };
+  const onClickConfirm = useCallback(
```

Comment:
should we rename this just to onConfirm? Does it matter if its a click or a touch?